src/main/java/com/nexenio/CppModernJson/CppModernJsonCodegen.java

```diff
@@ -207,11 +209,25 @@ public String toModelFilename(String name) {
   public String getSchemaType(Schema p) {
     String openAPIType = super.getSchemaType(p);
     String type = null;
+
+    Map<String, Schema> allDefinitions = ModelUtils.getSchemas(this.openAPI);
+    Schema completeSchema = allDefinitions.get(openAPIType);
+
     if (typeMapping.containsKey(openAPIType)) {
       type = typeMapping.get(openAPIType);
-      if (languageSpecificPrimitives.contains(type)) return toModelName(type);
-    } else type = openAPIType;
-    return toModelName(type);
+
+      if (languageSpecificPrimitives.contains(type)) {
+        return toModelName(type);
+      }
+    } else {
+      type = openAPIType;
+    }
+
+    if(completeSchema != null && completeSchema.getDiscriminator() != null) {
```

Comment:
This applies to anything with a discriminator, right? We'll probably have to change this when we get around to `oneOf`.

Reply:
Yes, that's very possible. I would like to leave it for now, though